cmd/gardener-extension-provider-aws/app/migrations.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/go-logr/logr"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var nameRegex = regexp.MustCompile("extensions.gardener.cloud:provider-aws:shoot--.*:machine-controller-manager")
+
+// TODO (georgibaltiev): Remove after the release of version 1.62.0
+func purgeMachineControllerManagerRBACResources(ctx context.Context, c client.Client, log logr.Logger) error {
+	log.Info("Starting the removal of obsolete ClusterRoles and ClusterRoleBindings")
+
+	var (
+		clusterRoleBindingList = &rbacv1.ClusterRoleBindingList{}
+		clusterRoleList        = &rbacv1.ClusterRoleList{}
+	)
+
+	if err := c.List(ctx, clusterRoleBindingList); err != nil {
+		return fmt.Errorf("failed to list ClusterRoleBindings: %w", err)
+	}
+
+	for _, clusterRoleBinding := range clusterRoleBindingList.Items {
+		if nameRegex.Match([]byte(clusterRoleBinding.Name)) {
+			log.Info("Removing ClusterRoleBinding", "ClusterRoleBinding", client.ObjectKeyFromObject(&clusterRoleBinding))
```

Comment:
From https://github.com/gardener/gardener/blob/v1.118.0/docs/development/logging.md#keys-and-values:

> Use [lowerCamelCase](https://en.wiktionary.org/wiki/lowerCamelCase) for keys. Don't put spaces in keys, as it will make log processing with simple tools like jq harder.

```suggestion
			log.Info("Deleting clusterRoleBinding", "ClusterRoleBinding", client.ObjectKeyFromObject(&clusterRoleBinding))
```